extensions/azurecore/src/account-provider/tokenCache.ts

```diff
@@ -32,12 +32,32 @@ export default class TokenCache implements adal.TokenCache {
 				.then(cache => self.addToCache(cache, entries))
 				.then(updatedCache => self.writeCache(updatedCache))
 				.then(
-					() => callback(null, false),
-					(err) => callback(err, true)
+					() => callback(null, true),
```

Comment:
Why is this even implemented as a callback function? is there some dependency that requires this? promises are cleaner.

Reply:
This is unfortunately part of the adal.TokenCache interface. https://github.com/AzureAD/azure-activedirectory-library-for-nodejs/blob/dev/lib/adal.d.ts#L169

I could move the actual logic into its own promise-implemented functions and then have the overrides just call those but that's adding more code for no real gain so I'm just going to leave it as is - although I agree with you that this is pretty messy to look at.